_episodes/01-introduction.md

```diff
@@ -4,14 +4,105 @@ teaching: 15
 exercises: 0
 questions:
 - "What is programming?"
-- "How to document code?"
-- "Finding resources/code online"
+- "How do I document code?"
+- "How do I find reliable and safe resources or code online?"
 objectives:
 - "identify basic concepts in programming"
 keypoints:
 - 
 ---
-FIXME
+
+## Programming in Python
+
+In most general terms, programming is the process of writing instructions for a computer. In this course we will be using Python as the language to communicate with the computer. 
+
+> Strictly speaking, Python is an interpreted language, rather than a compiled language, meaning we are not communicating directly with the computer when we use Python.
+
+Programming is a wide topic including a variety of techniques and tools. In this course we'll be focusing on programming for statistical analysis.
+
+
+### IDEs
+
+IDE stands for Integrated Development Environment. IDEs are where you will write, edit, and debug python scripts, so you want to choose one that makes you feel comfortable and includes the functionality that you need. 
+
+### Packages
+
+Packages, or libraries, are extensions to the statistical programming language. They contain code, data, and documentation in a standardised collection format that can be installed by users, typically via a centralised software repository.
+
+## Best Practices
+
+Let's overview some base concepts that any programmer should always keep in mind.
+
+### Documentation
+
+Have you ever returned to a task and tried to read a note that you quickly scrawled for yourself the last time you were working on it? Have you ever inherited a project from a colleague and foudn you have no idea what remains to be done?
```

Comment:
Have you ever returned to a task and tried to read a note that you quickly scrawled for yourself the last time you were working on it? Have you ever inherited a project from a colleague and **found** you have no idea what remains to be done?